feat: Add support for dependency dashboards in Azure DevOps

VNovotna opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

<!-- Describe what behavior is changed by this PR. -->
This PR implements read-only Dependency Dashborads in Azure DevOps platform. It uses the new option to have issue body in markdown. The dependency dashboards are read-only as it seems Azure DevOps strips HTML-like comments from issue body. I did not try to work around it now.

### Implementation remarks
* Azure DevOps has issues (called work items) per project not per repository, thus prefix `[repo-name]` is added to Dependency dashboard title
* Issue manipulation produces only warnings on failure. This is to avoid BC break as PATs do not require Read & write access to Work Items by current documentation.
* There is also expanded readOnlyIssueBody() method, to also remove checkboxes for Config migration PRs. As those check-boxes are also ignored.

## Context

Please select one of the below:

- [x] This closes an existing Issue: #9592 
- [ ] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

<!-- We request this information to assist reviewers in identifying AI-generated errors and other issues specific to AI usage. While we typically permit the use of AI tools, we appreciate being notified when they are employed. -->

Did you use AI tools to create any part of this pull request?

Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.

- [ ] No — I did not use AI for this contribution.
- [ ] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes — substantive assistance (AI generated non‑trivial portions of code, tests, or documentation).
- [x] Yes — other (please describe): AI was used to analyze other platforms implementation and suggest implementation for Azure DevOps. It was also used for multi-line completions, predominantly in test files. 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

The public repository: Sorry this is something I can not provide. More so, given the nature of Azure DevOps, access to issues is needed instead. I know screenshot is poor replacement, but butter than nothing I guess.
<img width="1613" height="886" alt="Vložený obrázek" src="https://github.com/user-attachments/assets/d066c8d6-c9f5-4efe-a9e1-e8bee92f9366" />



<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

VNovotna committed:
Add support for Azure DevOps in dependency dashboards feature

related #9592

VNovotna committed:
Add missing test for got wrapper

RahulGautamSingh commented on lib/modules/platform/azure/index.ts:
```diff
@@ -857,48 +860,231 @@ export async function mergePr({
+      .replace(regEx(/<!--renovate-(?:debug|config-hash):.*?-->/g), '')
+      // Replace GitHub-style PR links with Azure DevOps format
+      .replace(regEx(/\]\(\.\.\/pull\//g), '](!')
+      // Replace GitHub-style PR references (#123) with Azure DevOps format, needed for text linking migration config PR
```
```suggestion
      // Replace GitHub-style PR references (#123) with Azure DevOps format, needed for text linking config migration PR
```

RahulGautamSingh commented on lib/modules/platform/azure/util-workitem.ts:
move it to utils file

VNovotna committed:
Apply all current CR suggestions

VNovotna commented on lib/modules/platform/azure/index.ts:
```diff
@@ -857,48 +860,231 @@ export async function mergePr({
-    )
-    .replace(regEx(`\n---\n\n.*?<!-- rebase-check -->.*?\n`), '')
-    .replace(regEx(/<!--renovate-(?:debug|config-hash):.*?-->/g), '');
+  return (
```
Hi, `pnpm lint-fix` keeps adding them, should I try to make linter ignore this?

sanderjochems-capgemini reviewed (approved):
LGTM

sanderjochems-capgemini commented:
@VNovotna I see there are some merge conflict with the base branch now

RahulGautamSingh reviewed (commented):
@VNovotna Are you able to see this PR through?

viceice reviewed (commented):
conflicted

VNovotna committed:
Merge upstream/main into azdo-dep-dashboards

Resolved conflicts in:
- lib/modules/platform/azure/azure-got-wrapper.ts
- lib/modules/platform/azure/index.ts
- lib/modules/platform/azure/types.ts
- lib/modules/platform/utils/read-only-issue-body.ts

VNovotna committed:
fix: resolve eslint and tsx issues in Azure platform code

VNovotna committed:
test: add coverage for issue.number undefined cases

Add test cases to ensure proper handling when:
- Trying to reopen an issue without a number
- Trying to update an issue without a number

This improves code coverage for the null checks added in previous commit.

VNovotna committed:
test: fix and add tests of azure/issue.ts to up coverage

there was mismatch in mock resulting in testing of different code branch

xrow commented:
@VNovotna Cool Feature. Thanks. Can`t await it.

RahulGautamSingh commented:
Due to inactivity I am reviving this pr https://github.com/renovatebot/renovate/pull/42758, as the required changes are very minimal to just close it.
